toolchain/install/install_paths.h

```diff
@@ -0,0 +1,78 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef CARBON_TOOLCHAIN_INSTALL_INSTALL_PATHS_H_
+#define CARBON_TOOLCHAIN_INSTALL_INSTALL_PATHS_H_
+
+#include "common/ostream.h"
+#include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/StringRef.h"
+
+namespace Carbon {
+
+// Locates the toolchain installation and provides paths to various components.
+//
+// The Carbon toolchain expects to be installed into some install prefix. For
+// example, this is expected to be similar to the CMake install prefix:
+//
+// - `C:/Program Files/Carbon` or similar on Windows.
+// - `/usr` or `/usr/local` on Linux and most BSDs.
+// - `/opt/homebrew` or similar on macOS with Homebrew.
+// - `bazel-bin/some/bazel/target.runfiles/_main/toolchain/install/prefix_root`
+//   for unit tests and just-built binaries during development.
+//
+// See https://cmake.org/cmake/help/latest/variable/CMAKE_INSTALL_PREFIX.html
+// for more details. While we don't build the toolchain with CMake, we expect
+// our installation to behave in a similar and compatible way.
+//
+// We detect the installation prefix by looking at the path of the executable.
+// This supports development binaries like tests, as well as any installed
+// user-facing binaries.
+//
+// Within this prefix, we expect a hierarchy on Unix-y platforms:
+//
+// - `prefix_root/bin/carbon` - the main CLI driver.
+// - `prefix_root/lib/carbon` - private data & binaries used by the toolchain.
```

Comment:
```suggestion
// - `prefix_root/lib/carbon/...` - private data & binaries used by the toolchain.
```

```suggestion
// - `prefix_root/lib/carbon/` - private data & binaries used by the toolchain.
```

Stray thought, to distinguish the above line from this one. Not sure if the `...` is needed if you're trying to fit this, but I think the slash at least helps.